src/components/Account-panel/Contact-info/Contact-info.tsx

```diff
@@ -1,35 +1,96 @@
-import { useEffect, useState } from 'react'
-import { Form, Formik } from 'formik'
-import { Box, Grid, Typography } from '@mui/material'
-import axiosInstance from '../../../axios/settings'
+import { useState } from 'react'
+import {
+  Box,
+  FormHelperText,
+  Grid,
+  InputLabel,
+  Typography
+} from '@mui/material'
 import { stH3, stP } from '../../auth/style'
-import CustomInput from '../../auth/InputCustom'
 import { useDispatch } from 'react-redux'
 import { currentUser } from '../../../redux/authentication/operation'
 import { getToken } from '../../../utils/cookie/token'
 import { AppDispatch } from '../../../redux/store'
 import { EmailConfirmationModal } from '../../Modal-custom-btn/ModalCustomBtnEmail'
 import { UserResponse } from '../../../types'
 import { ChangeDataSchema } from '../../auth/validation'
-import { BootstrapButton } from './style'
 import { CustomSnackbar } from '../../SnackebarCustom/SnackbarCustom'
 import { useAuth } from '../../../hooks/use-auth'
+import { OutlinedInput } from '../../UI/Field'
+import { stBtn, stInput } from '../Change-password/Change-password'
+import { Button } from '../../UI/Button'
+import { SubmitHandler, UseFormRegister, useForm } from 'react-hook-form'
+import { yupResolver } from '@hookform/resolvers/yup'
+import axiosInstance from '../../../axios/settings'
 
-export type UserSubset = Pick<
-  UserResponse,
-  'email' | 'first_name' | 'last_name' | 'phone_number'
->
+type FormData = {
+  first_name: string
+  last_name: string
+  phone_number?: string | null
+  email?: string | undefined
+}
 
-export type SnackbarData = {
+type SnackbarData = {
   open: boolean
   error: boolean
-  message?: string | undefined
+  message?: string
+}
+
+type ContactInfoProps = {
+  user: UserResponse
 }
 
+type InputId = 'first_name' | 'last_name' | 'email' | 'phone_number'
+
+const renderInput = (
```

Comment:
remove all `render_something` this is a bad approach. If you have duplication -> move it as a component and re-use

Reply:
done